recipes/graphthewy/all/test_package/CMakeLists.txt

```
@@ -0,0 +1,10 @@
+cmake_minimum_required(VERSION 3.1)
+project(PackageTest CXX)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
+
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+add_executable(test_package test_package.cpp)
```

```suggestion
add_executable(test_package test_package.cpp)
target_link_libraries(test_package ${CONAN_LIBS})
```

It works without due to conan_basic_setup() and because your lib is header-only, but it's consistent with others recipes.

---

Thank you for the explanation, however it does not build locally without setting `-s compiler.cppstd=17` to `conan create` command. Am I missing a requirement, especially on the recipe ?

---

oh yes, you must also add `set_property(TARGET test_package PROPERTY CXX_STANDARD 17)`